src/lightning_wallet.ts

```diff
@@ -0,0 +1,180 @@
+import {
+  AddressType,
+  GetInfoResponse,
+  Invoice,
+  OpenStatusUpdate,
+  SendResponse
+} from "@radar/lnrpc";
+import pEvent from "p-event";
+import { Lnd } from "./lnd";
+
+export class LightningWallet {
+  public static async newInstance(
+    certPath: string,
+    macaroonPath: string,
+    lndHost: string,
+    lndPort: number
+  ): Promise<LightningWallet> {
+    const lnd = await Lnd.init({
+      tls: certPath,
+      macaroonPath,
+      server: `${lndHost}:${lndPort}`
+    });
+    return new LightningWallet(lnd, lndHost, lndPort);
+  }
+
+  public constructor(
+    public readonly lnd: Lnd,
+    private readonly p2pHost: string,
+    private readonly p2pPort: number
+  ) {}
+
+  public async sendPayment(
+    publicKey: string,
+    satAmount: string,
+    secretHash: string,
+    finalCltvDelta: number
+  ): Promise<SendResponse> {
+    const publicKeyBuf = Buffer.from(publicKey, "hex");
+    const paymentHash = Buffer.from(secretHash, "hex");
+
+    return this.lnd.lnrpc.sendPaymentSync({
+      dest: publicKeyBuf,
+      amt: satAmount,
+      paymentHash,
+      cltvLimit: finalCltvDelta
+    });
+  }
+
+  public async addHoldInvoice(
+    satAmount: string,
+    secretHash: string,
+    expiry: number,
+    memo: string
+  ): Promise<string> {
+    const satAmountNum = parseInt(satAmount, 10);
+
+    return (
+      await this.lnd.invoicesrpc.addHoldInvoice({
+        value: satAmountNum,
+        hash: secretHash,
+        expiry,
+        memo
+      })
+    ).paymentRequest;
+  }
+
+  public async settleInvoice(secret: string): Promise<void> {
+    await this.lnd.invoicesrpc.settleInvoice({ preimage: secret });
+  }
+
+  public async newAddress(type: AddressType): Promise<string> {
+    return (await this.lnd.lnrpc.newAddress({ type })).address;
+  }
+
+  public async confirmedChannelBalance(): Promise<string> {
```

Comment:
Which channel is this? On an API of this abstraction level, should we not pass a channel identifier?

Reply:
The use of the API is correct here, you cannot specify the channel:

```
    /**
     * channelBalance returns the total funds available across all open channels in satoshis.
     */
    channelBalance(args?: {}): Promise<ChannelBalanceResponse>;
```